libr/debug/dreg.c

```diff
@@ -128,9 +129,27 @@ R_API int r_debug_reg_list(RDebug *dbg, int type, int size, int rad, const char
 
 	int itmidx = -1;
 	dbg->creg = NULL;
-	head = r_reg_get_list (dbg->reg, type);
-	if (!head) {
-		return false;
+	if (type == R_REG_TYPE_ALL) {
+		head = r_list_new ();
+		for (i = 0; i < R_REG_TYPE_LAST; i++) {
+			RList *tmp = r_reg_get_list (dbg->reg, i);
+			if (!tmp || !r_list_length (tmp)) {
+				continue;
+			}
+			tmp = r_list_clone (tmp);
+			if (!r_list_join (head, tmp)) {
+				r_list_free (head);
+				r_list_free (tmp);
+				return false;
```

Comment:
this function returns int, not bool. so it will be good to change the return type to `bool` and fix the rest of return statemnts to return n != 0 instead of n